creator-node/src/contentAccess/contentAccessChecker.ts

```diff
@@ -0,0 +1,88 @@
+import { isRegisteredDiscoveryNode } from './helpers'
+import { CheckAccessArgs, CheckAccessResponse } from './types'
+import { recoverWallet, signatureHasExpired } from '../apiSigning'
+
+const PREMIUM_CONTENT_SIGNATURE_MAX_TTL_MS = 6 * 60 * 60 * 1000 // 6 hours
+
+/**
+ * Checks that all premium content headers are passed in.
+ * Checks that discovery node that generated signature is registered.
+ * Checks that signature is not too old.
+ * Checks that user requesting the content has access.
+ */
+export async function checkContentAccess({
+  cid,
+  contentAccessHeaders,
+  libs,
+  logger,
+  redis
+}: CheckAccessArgs): Promise<CheckAccessResponse> {
+  if (!contentAccessHeaders) {
+    return {
+      doesUserHaveAccess: false,
+      error: 'MissingHeaders',
+      shouldCache: false
+    }
+  }
+
+  const { signedDataFromDiscoveryNode, signatureFromDiscoveryNode } =
+    JSON.parse(contentAccessHeaders)
+  if (!signedDataFromDiscoveryNode || !signatureFromDiscoveryNode) {
+    return {
+      doesUserHaveAccess: false,
+      error: 'MissingHeaders',
+      shouldCache: false
+    }
+  }
+
+  const discoveryNodeWallet = recoverWallet(
+    signedDataFromDiscoveryNode,
+    signatureFromDiscoveryNode
+  )
+  const isRegisteredDN = await isRegisteredDiscoveryNode({
+    wallet: discoveryNodeWallet,
+    libs,
+    logger,
+    redis
+  })
+  if (!isRegisteredDN) {
+    return {
+      doesUserHaveAccess: false,
+      error: 'InvalidDiscoveryNode',
```

Comment:
nit maybe can call this `InvalidSignature` or `InvalidDiscoverySignature` or smth since we don't know if the discoverynode is invalid or if the signature was just poorly formatted etc